subsys/usb/device_next/class/usbd_cdc_acm.c

```diff
@@ -394,14 +396,36 @@ static int usbd_cdc_acm_ctd(struct usbd_class_node *const c_nd,
 static int usbd_cdc_acm_init(struct usbd_class_node *const c_nd)
 {
 	struct usbd_cdc_acm_desc *desc = c_nd->data->desc;
+	const struct device *dev = c_nd->data->priv;
+	struct cdc_acm_uart_data *data = dev->data;
 
 	desc->iad_cdc.bFirstInterface = desc->if0.bInterfaceNumber;
 	desc->if0_union.bControlInterface = desc->if0.bInterfaceNumber;
 	desc->if0_union.bSubordinateInterface0 = desc->if1.bInterfaceNumber;
 
+	if (usbd_add_descriptor(c_nd->data->uds_ctx, data->iad_desc_nd)) {
+		LOG_ERR("Failed to add iFunction string descriptor");
+	} else {
+		desc->iad_cdc.iFunction = usbd_get_descriptor_idx(data->iad_desc_nd);
+		desc->if0.iInterface = usbd_get_descriptor_idx(data->iad_desc_nd);
+		desc->if1.iInterface = usbd_get_descriptor_idx(data->iad_desc_nd);
```

Comment:
I don't quite like using the same string for all three. I think only the iFunction should be set here.

Reply:
That was my first thought too, but I am afraid there are users who expect it (also) at interface level.

Reply:
I think the users would expect to be able to set some meaningful interface string descriptors. Having the same string descriptor assigned to both interfaces and function is not really making things better IMHO.